spdx/common/identifier.go

```diff
@@ -8,12 +8,58 @@ import (
 	"strings"
 )
 
+const (
+	spdxRefPrefix     = "SPDXRef-"
+	documentRefPrefix = "DocumentRef-"
+)
+
 // ElementID represents the identifier string portion of an SPDX element
 // identifier. DocElementID should be used for any attributes which can
 // contain identifiers defined in a different SPDX document.
 // ElementIDs should NOT contain the mandatory 'SPDXRef-' portion.
 type ElementID string
 
+// MarshalJSON returns an SPDXRef- prefixed JSON string
+func (d ElementID) MarshalJSON() ([]byte, error) {
+	return json.Marshal(prefixElementId(d))
+}
+
+// UnmarshalJSON validates SPDXRef- prefixes and removes them when processing ElementIDs
+func (d *ElementID) UnmarshalJSON(data []byte) error {
+	// SPDX identifier will simply be a string
+	idStr := string(data)
+	idStr = strings.Trim(idStr, "\"")
+
+	e, err := trimElementIdPrefix(idStr)
+	if err != nil {
+		return err
+	}
```

Comment:
NOTE: this will mean parsing invalid SPDX files will fail due to invalid SPDX identifiers; unsure if that should be the behavior here. It is consistent with the original `DocElementID.UnmarshalJSON` behavior: https://github.com/spdx/tools-golang/pull/170/files#diff-c46b7836ea8c59261500de448b948294dc3d818d706da8b08dbed529bd1a9609L71

Reply:
i would prefer to be stricter here, so this should be fine.